src/pint/utils.py

```diff
@@ -1552,3 +1558,98 @@ def remove_dummy_distance(c):
             "Do not know coordinate frame for %r: returning coordinates unchanged" % c
         )
         return c
+
+
+def calculate_random_models(fitter, toas, Nmodels=100, keep_models=True, params="all"):
+    """
+    calculates random models based on the covariance matrix of the `fitter` object
+    
+    returns the new phase differences compared to the original model
+    optionally returns all of the random models
+
+    Parameters
+    ----------
+    fitter: `pint.fitter` object
+        current fitter object containing a model and parameter covariance matrix
+    toas: `pint.toa.TOAs` object
+        TOAs to calculate models
+    Nmodels: int (optional)
+        number of random models to calculate
+    keep_models: bool (optional)
+        whether to keep and return the individual random models (slower)
+    params: list (optional)
+        if specified, selects only those parameters to vary.  Default ('all') is to use all parameters other than Offset
+
+    Returns
+    -------
+    dphase : np.ndarray
+        phase difference with respect to input model, size is [Nmodels, len(toas)]
+    random_models : list (optional)
+        list of random models (each is a `pint.models.timing_model.TimingModel`)
+
+
+    Note: to calculate new TOAs, you can do:
```

Comment:
There's a format for a Note section (and for embedded source code).

Reply:
fixed.